packages/core/src/errors.ts

```diff
@@ -12,9 +12,22 @@ export enum ProviderFailure {
 
 const formatMessage = (reason: string, detail?: string) => reason + (detail ? ` (${detail})` : '');
 
+const tryToUnhideInnerError = (error: Error, innerError: unknown) => {
+  if (!(innerError instanceof Error) || !error.stack || !innerError.stack) return;
+
+  const [firstLineOfErrorStack] = error.stack.split('\n    at ');
+
+  Object.defineProperty(error, 'stack', {
+    configurable: true,
+    value: `${firstLineOfErrorStack} due to \n ${innerError.stack}`
+  });
+};
+
 export class ProviderError<InnerError = unknown> extends CustomError {
   constructor(public reason: ProviderFailure, public innerError?: InnerError, public detail?: string) {
     super(formatMessage(reason, detail));
```

Comment:
Would be great to also concatenate the message

Reply:
It is @mkazlauskas : the value of the `stack` property is the concatenation of `type` + `message` + `stack trace`

Reply:
Would concatenating type/message/stack trace individually work then? :thinking:

Reply:
Sorry I'm afraid I didn't got the point.

Any solution _works_... I can't understand which is the target you want to reach... which is the problem you see an you are trying to resolve.

Reply:
From your comment it seems like 'stack' is a computed property. If that's the case, then overwriting the source of this property seems like a better way to do this, because then it would behave consistently with other Error objects

Reply:
Anyway, current approach works, let's not overthink it. Marking this as resolved.